mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/DefaultMapperContext.java

```diff
@@ -73,7 +92,19 @@ public DefaultMapperContext withKeyspaceAndTable(
             && Objects.equals(newTableId, this.tableId))
         ? this
         : new DefaultMapperContext(
-            session, newKeyspaceId, newTableId, nameConverterCache, customState);
+            session, newKeyspaceId, newTableId, null, nameConverterCache, customState);
+  }
+
+  public DefaultMapperContext withExecutionProfile(@Nullable String newExecutionProfile) {
+    return newExecutionProfile.equals(this.executionProfile)
```

Comment:
IDEA warning: Method invocation 'equals' may produce 'NullPointerException'

You can use `Objects.equals()` instead.